frontend/src/components/SignupForm.tsx

```diff
@@ -24,9 +26,36 @@ export default function SignupForm({
   const location = useStatefulLocation()
   const { register } = useUser()
   const { from } = location.state || { from: { pathname: "/" } }
+  const [isModalOpen, setIsModalOpen] = React.useState<boolean>(false)
+  const [isNotifOpen, setNotifOpen] = React.useState<boolean>(false)
+  const [type] = React.useState<NotificationType["type"]>("error")
+  const [message] = React.useState("Something went wrong!")
```

Comment:
These states never change and can therefore just be constant.
```suggestion
  const type: NotificationType["type"] = "error"
  const message = "Something went wrong!"
```